### ISSUE TITLE
Broken document build 1.12.0

### ISSUE THREAD

**Issue by facelessuser** [squidfunk/mkdocs-material#522]
## Description

Cannot build with 1.12.0 as it produces Jinja2 errors.

```
0.74s$ mkdocs build --clean --verbose --strict
DEBUG - Loading configuration file: /home/travis/build/[secure]/sublime-markdown-popups/mkdocs.yml
DEBUG - Config value: 'remote_name' = 'origin'
DEBUG - Config value: 'include_next_prev' = True
DEBUG - Config value: 'extra_css' = ['extra-9b4c72abee.css']
DEBUG - Config value: 'repo_name' = 'GitHub'
DEBUG - Config value: 'mdx_configs' = {'pymdownx.escapeall': {'nbsp': True, 'hardbreak': True}, 'pymdownx.keys': {'separator': '＋'}, 'pymdownx.tasklist': {'custom_checkbox': True}, 'pymdownx.snippets': {'base_path': 'docs/src/markdown/_snippets'}, 'pymdownx.magiclink': {'repo_url_shortener': True, 'base_repo_url': 'https://github.com/[secure]/sublime-markdown-popups'}, 'markdown.extensions.toc': {'slugify': <function uslugify at 0x7f2d321f4b00>, 'permalink': '\ue157'}, 'markdown.extensions.smarty': {'smart_quotes': False}, 'pymdownx.emoji': {'emoji_generator': <function to_png at 0x7f2d32b3c8c0>}, 'pymdownx.plainhtml': {'strip_attributes': ''}, 'pymdownx.highlight': {'css_class': 'codehilite', 'extend_pygments_lang': [{'name': 'php-inline', 'lang': 'php', 'options': {'startinline': True}}]}}
DEBUG - Config value: 'extra_javascript' = ['extra-449b50d2b3.js']
DEBUG - Config value: 'repo_url' = 'https://github.com/[secure]/sublime-markdown-popups'
DEBUG - Config value: 'site_url' = 'https://[secure].github.io/sublime-markdown-popups'
DEBUG - Config value: 'site_name' = 'Sublime Markdown Popups Documentation'
DEBUG - Config value: 'edit_uri' = 'tree/master/docs/src/markdown'
DEBUG - Config value: 'use_directory_urls' = True
DEBUG - Config value: 'mkdocs_templates' = '/home/travis/virtualenv/python3.3.6/lib/python3.3/site-packages/mkdocs/templates'
DEBUG - Config value: 'remote_branch' = 'gh-pages'
DEBUG - Config value: 'extra' = {'palette': {'primary': 'blue', 'accent': 'blue'}, 'font': {'text': 'Roboto', 'code': 'Roboto Mono'}, 'social': [{'link': 'https://github.com/[secure]', 'type': 'github'}]}
DEBUG - Config value: 'markdown_extensions' = ['meta', 'toc', 'tables', 'fenced_code', 'markdown.extensions.toc', 'markdown.extensions.admonition', 'markdown.extensions.smarty', 'pymdownx.betterem', 'markdown.extensions.attr_list', 'markdown.extensions.def_list', 'markdown.extensions.tables', 'markdown.extensions.abbr', 'pymdownx.extrarawhtml', 'pymdownx.superfences', 'pymdownx.highlight', 'pymdownx.inlinehilite', 'pymdownx.magiclink', 'pymdownx.tilde', 'pymdownx.caret', 'pymdownx.smartsymbols', 'pymdownx.emoji', 'pymdownx.escapeall', 'pymdownx.tasklist', 'pymdownx.progressbar', 'pymdownx.plainhtml', 'pymdownx.snippets', 'pymdownx.keys', 'pymdownx.details']
DEBUG - Config value: 'strict' = True
DEBUG - Config value: 'dev_addr' = '127.0.0.1:8000'
DEBUG - Config value: 'site_dir' = '/home/travis/build/[secure]/sublime-markdown-popups/site'
DEBUG - Config value: 'docs_dir' = '/home/travis/build/[secure]/sublime-markdown-popups/docs/src/markdown'
DEBUG - Config value: 'extra_templates' = []
DEBUG - Config value: 'copyright' = 'Copyright &copy; 2015 - 2017 <a href="https://github.com/[secure]">Isaac Muse</a>'
DEBUG - Config value: 'include_nav' = True
DEBUG - Config value: 'theme' = 'material'
DEBUG - Config value: 'site_author' = None
DEBUG - Config value: 'site_description' = 'Popup tooltips for Sublime generated with Markdown.'
DEBUG - Config value: 'config_file_path' = '/home/travis/build/[secure]/sublime-markdown-popups/mkdocs.yml'
DEBUG - Config value: 'site_favicon' = None
DEBUG - Config value: 'theme_dir' = ['/home/travis/build/[secure]/sublime-markdown-popups/docs/theme', '/home/travis/virtualenv/python3.3.6/lib/python3.3/site-packages/material', '/home/travis/virtualenv/python3.3.6/lib/python3.3/site-packages/mkdocs/assets/search']
DEBUG - Config value: 'pages' = [{'About Markdown Popups': 'index.md'}, {'Installation': 'installation.md'}, {'User Settings': 'settings.md'}, {'API': 'api.md'}, {'CSS Styling': 'styling.md'}, {'F.A.Q': 'faq.md'}, {'Contributing &amp; Support': 'contributing.md'}, {'Changelog': 'changelog.md'}, {'License': 'license.md'}]
DEBUG - Config value: 'google_analytics' = None
INFO - Cleaning site directory
INFO - Building documentation to directory: /home/travis/build/[secure]/sublime-markdown-popups/site
DEBUG - Copying static assets from theme: /home/travis/virtualenv/python3.3.6/lib/python3.3/site-packages/mkdocs/assets/search
DEBUG - Copying static assets from the docs dir.
DEBUG - Building markdown pages.
DEBUG - Building template: 404.html
Traceback (most recent call last):
 File "/home/travis/virtualenv/python3.3.6/bin/mkdocs", line 11, in <module>
 sys.exit(cli())
 File "/home/travis/virtualenv/python3.3.6/lib/python3.3/site-packages/click/core.py", line 722, in __call__
 return self.main(*args, **kwargs)
 File "/home/travis/virtualenv/python3.3.6/lib/python3.3/site-packages/click/core.py", line 697, in main
 rv = self.invoke(ctx)
 File "/home/travis/virtualenv/python3.3.6/lib/python3.3/site-packages/click/core.py", line 1066, in invoke
 return _process_result(sub_ctx.command.invoke(sub_ctx))
 File "/home/travis/virtualenv/python3.3.6/lib/python3.3/site-packages/click/core.py", line 895, in invoke
 return ctx.invoke(self.callback, **ctx.params)
 File "/home/travis/virtualenv/python3.3.6/lib/python3.3/site-packages/click/core.py", line 535, in invoke
 return callback(*args, **kwargs)
 File "/home/travis/virtualenv/python3.3.6/lib/python3.3/site-packages/mkdocs/__main__.py", line 156, in build_command
 ), dirty=not clean)
 File "/home/travis/virtualenv/python3.3.6/lib/python3.3/site-packages/mkdocs/commands/build.py", line 380, in build
 build_pages(config, dirty=dirty)
 File "/home/travis/virtualenv/python3.3.6/lib/python3.3/site-packages/mkdocs/commands/build.py", line 307, in build_pages
 build_template('404.html', env, config, site_navigation)
 File "/home/travis/virtualenv/python3.3.6/lib/python3.3/site-packages/mkdocs/commands/build.py", line 167, in build_template
 output_content = template.render(context)
 File "/home/travis/virtualenv/python3.3.6/lib/python3.3/site-packages/jinja2/environment.py", line 1008, in render
 return self.environment.handle_exception(exc_info, True)
 File "/home/travis/virtualenv/python3.3.6/lib/python3.3/site-packages/jinja2/environment.py", line 780, in handle_exception
 reraise(exc_type, exc_value, tb)
 File "/home/travis/virtualenv/python3.3.6/lib/python3.3/site-packages/jinja2/_compat.py", line 37, in reraise
 raise value.with_traceback(tb)
 File "/home/travis/virtualenv/python3.3.6/lib/python3.3/site-packages/material/404.html", line 1, in top-level template code
 {% extends "base.html" %}
 File "/home/travis/virtualenv/python3.3.6/lib/python3.3/site-packages/material/base.html", line 162, in top-level template code
 {% block footer %}
 File "/home/travis/virtualenv/python3.3.6/lib/python3.3/site-packages/material/base.html", line 163, in block "footer"
 {% include "partials/footer.html" %}
 File "/home/travis/build/[secure]/sublime-markdown-popups/docs/theme/partials/footer.html", line 1, in top-level template code
 {% import "partials/language.html" as lang %}
 File "/home/travis/virtualenv/python3.3.6/lib/python3.3/site-packages/jinja2/environment.py", line 1073, in make_module
 return TemplateModule(self, self.new_context(vars, shared, locals))
 File "/home/travis/virtualenv/python3.3.6/lib/python3.3/site-packages/jinja2/environment.py", line 1152, in __init__
 body_stream = list(template.root_render_func(context))
 File "/home/travis/virtualenv/python3.3.6/lib/python3.3/site-packages/material/partials/language.html", line 1, in top-level template code
 {% set languages = config.extra.language | default("en") %}
 File "/home/travis/virtualenv/python3.3.6/lib/python3.3/site-packages/jinja2/environment.py", line 430, in getattr
 return getattr(obj, attribute)
jinja2.exceptions.UndefinedError: 'config' is undefined
```
### Expected behavior

Build without errors.

### Actual behavior

Cannot build and get aforementioned error.

### Steps to reproduce the bug

1. Run command above and get the above error.

### Package versions

* Python: 3.3 in the above example
* MkDocs: 0.17
* Material: 1.12.0

### Project configuration

``` yaml
site_name: Sublime Markdown Popups Documentation
site_url: https://facelessuser.github.io/sublime-markdown-popups
repo_url: https://github.com/facelessuser/sublime-markdown-popups
edit_uri: tree/master/docs/src/markdown
site_description: Popup tooltips for Sublime generated with Markdown.
copyright: Copyright &copy; 2015 - 2017 <a href="https://github.com/facelessuser">Isaac Muse</a>

pages:
  - About Markdown Popups: index.md
  - Installation: installation.md
  - User Settings: settings.md
  - API: api.md
  - CSS Styling: styling.md
  - F.A.Q: faq.md
  - Contributing &amp; Support: contributing.md
  - Changelog: changelog.md
  - License: license.md

theme: material
docs_dir: docs/src/markdown
theme_dir: docs/theme

markdown_extensions:
  - markdown.extensions.toc:
      slugify: !!python/name:pymdownx.slugs.uslugify
      permalink: "\ue157"
  - markdown.extensions.admonition:
  - markdown.extensions.smarty:
      smart_quotes: false
  - pymdownx.betterem:
  - markdown.extensions.attr_list:
  - markdown.extensions.def_list:
  - markdown.extensions.tables:
  - markdown.extensions.abbr:
  - pymdownx.extrarawhtml:
  - pymdownx.superfences:
  - pymdownx.highlight:
      css_class: codehilite
      extend_pygments_lang:
        - name: php-inline
          lang: php
          options:
            startinline: true
  - pymdownx.inlinehilite:
  - pymdownx.magiclink:
      repo_url_shortener: true
      base_repo_url: https://github.com/facelessuser/sublime-markdown-popups
  - pymdownx.tilde:
  - pymdownx.caret:
  - pymdownx.smartsymbols:
  - pymdownx.emoji:
      emoji_generator: !!python/name:pymdownx.emoji.to_png
  - pymdownx.escapeall:
      hardbreak: true
      nbsp: true
  - pymdownx.tasklist:
      custom_checkbox: true
  - pymdownx.progressbar:
  - pymdownx.plainhtml:
      strip_attributes: ''
  - pymdownx.snippets:
      base_path: docs/src/markdown/_snippets
  - pymdownx.keys:
      separator: "\uff0b"
  - pymdownx.details:

extra:
  palette:
    primary: blue
    accent: blue
  font:
    text: Roboto
    code: Roboto Mono
  social:
    - type: github
      link: https://github.com/facelessuser
extra_css:
  - extra-9b4c72abee.css
extra_javascript:
  - extra-449b50d2b3.js

```

### System information

* OS:  Linux above but most likely any
* Browser: NA

**Comment by squidfunk** [comment]
Did you override any partials?

**Comment by facelessuser** [comment]
&lt;facepalm&gt; yeah, that is exactly what I did.

**Comment by squidfunk** [comment]
I will provide downward compatibility

**Comment by facelessuser** [comment]
That's good too.  But I should really have checked that first, it's not the first time I've gotten burned by that.  You'd think I'd learn 🙂 .

**Comment by squidfunk** [comment]
Fixed in #523
